### PR TITLE
Upgrade Komodo Package to v0.5.4

### PR DESCRIPTION
**Pull request by parseccentric**
# Upgrade Komodo Package to v0.5.4

## Cleaned up and refactored code, adjusted colors

Commented out code related to Primitives and Grouping features, which will come later. Made ImpressMenu2 panels into prefabs. Adjusted colors.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Testing 
- [ ] Change that requires a documentation update

### How Has This Been Tested?

Visually inspect menu for coloring consistency.

## Repaired eraser

Fixed bug where eraser would only be enabled for the left hand.

See more details in #1 

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Testing 
- [ ] Change that requires a documentation update

### How Has This Been Tested?

Steps to test: 

1. Enter Play mode. 
2. Make sure the Scene and Game windows are visible at the same time
3. Select ManualTestHelper in the hierarchy.
4. Right click the script component on it and choose "Switch to XR"
5. Move handL so that it points at the Create tab
6. In the Game window, press the T key to open the Create panel
7. Use the same technique to press the Erase button

Expected result: both eraser displays show up.

- [x] Manual Test
- [ ] Unit Test
- [ ] Integration / End-to-End Test

## Added ManualTestHelper and ControllersManager

Created ManualTestHelper script and gameObject. To use it, right click the script component on the gameObject in the Main scene and select a test function.

I also introduced ControllersManager, which is a future candidate for refactoring KomodoControllerInteraction.cs. For now, it supports the `T` and `Y` keys, which correspond to the left controller trigger and the right controller trigger. To use this, make sure the game window is focused and the menu mode is set to XR mode.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Testing 
- [ ] Change that requires a documentation update

### How Has This Been Tested?

See the test for "Repaired Eraser" above.

Expected result: holding `T` activates left selection, drawing, or erasing, and holding `Y` activates right selection, drawing, or erasing

- [x] Manual Test
- [ ] Unit Test
- [ ] Integration / End-to-End Test

## Added ManualTestHelper and ControllersManager

Created ManualTestHelper script and gameObject. To use it, right click the script component on the gameObject in the Main scene and select a test function.

I also introduced ControllersManager, which is a future candidate for refactoring KomodoControllerInteraction.cs. For now, it supports the `T` and `Y` keys, which correspond to the left controller trigger and the right controller trigger. To use this, make sure the game window is focused and the menu mode is set to XR mode.

## Repaired ColorPicker color selection

Color selection now reflects the correct color rather than black or white. It also works on the first opening of the draw tab, unlike before. 

Under the hood, ColorPicker has been reworked into a prefab. Most functions have been moved to a static class, ColorPickerManager. The script component on the ColorImage has been reduced to the essentials. Both classes are initialized regardless of whether the draw tab is open thanks to a new script, CreateMenu.

See more details in #1 

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Testing 
- [x] Change that requires a documentation update

### How Has This Been Tested?

See the test for "Repaired Eraser" above for how to use the VR hands in the editor on a screen.

Expected result: selecting the draw tab results in no errors, the selected color is initially white, hovering over the ColorImage results in a correct preview color, and selecting a color results in the correctly displayed color, and drawing with that color results in the correct draw stroke color.

- [x] Manual Test
- [ ] Unit Test
- [ ] Integration / End-to-End Test
____

## Test Configuration

No browser yet
No VR headset yet

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas that are not self-documenting
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings 
- [ ] My changes have no unnecessary logging
- [ ] I have added tests that prove my fix is effective or that my feature works, for sufficiently complex features
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Sensitive info like tokens, secrets, and passwords have been removed before submitting

Modified from this article:
Phillip Johnston, “A GitHub Pull Request Template for Your Projects - Embedded Artistry,” Embedded Artistry, Aug. 04, 2017. https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ (accessed Jul. 22, 2021).

### CODE REVIEW

**Comment by parseccentric** [comment]
## Removed trusses, adjusted lighting

Removed large structures and softened lighting to nearly eliminate shadows.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Testing 
- [ ] Change that requires a documentation update

### How Has This Been Tested?

Visually inspect.

## Repaired UIManager bug, Repaired main scene

Fixed menu initialization sequence. Fixed main scene by replacing the missing SocketIOAdapter and ConnectionAdapter components.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Testing 
- [ ] Change that requires a documentation update

### How Has This Been Tested?

Enter play mode. Expected result: menu finishes initializing and loading overlay goes away. 

Enter a build. Connection status window should indicate connection status info from the relay server (or a poor connection).

**Comment by parseccentric** [comment]
@Barasakar When you have a chance, can you review and approve this and let me know?

File: KOMODO-IMPRESS/Assets/IMPRESS/Scripts/ColorPickerImage.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            if (RectTransformUtility.ScreenPointToLocalPointInRectangle(
+                rectTransform, globalPoint, eventCamera, out Vector2 localPoint))
+            {
+                float localX = localPoint.x / rectTransform.rect.width;
+
+                float localY = Mathf.Abs(localPoint.y / rectTransform.rect.height);
+
+                return new Vector2(localX, localY);
+            }
+
```
Comment (Barasakar) [inline KOMODO-IMPRESS/Assets/IMPRESS/Scripts/ColorPickerImage.cs:27]: spacing

Comment (Barasakar) [inline KOMODO-IMPRESS/Assets/IMPRESS/Scripts/ColorPickerImage.cs:27]: spacing

**Review by Barasakar — APPROVED** [review]

**Comment by parseccentric** [comment]
This addresses https://github.com/gelic-idealab/impress/issues/4

**Pull request merged**